psi4/src/psi4/libfock/v.cc

```diff
@@ -3326,15 +3335,36 @@ SharedMatrix UV::compute_gradient() {
         rhobyq[rank] += C_DDOT(npoints, QTp, 1, y, 1);
         rhobzq[rank] += C_DDOT(npoints, QTp, 1, z, 1);
 
-        // => LSDA Contribution <= //
+        // NOTE:
+        // Let d/dx be a derivative of a functon in R^3 with respect to coordinate x.
+        // Let d/d(x,i) be a derivative with respect to displacing atom i in the x direction.
+        // Then d/d(x,i) phi_mu,p = - d/dx phi^x_mu,p * δ_mu,i
+        //    δ = 1 if mu centered on atom i, else 0.
+        // i.e., the change in a basis function at a point if you move right is the same as the
+        // change if you move the basis function left.
+
+        // NOTE:
+        // ∂E/∂D ∂D/∂x = 0 because ∂E/∂D = 0 by Hellmann-Feynman.
+        // Therefore, we don't compute density-derivative terms. They sum to zero.
```

Comment:
This is not Hellmann-Feynman! It's just because we assume that the density has been optimized and the SCF equations are satisfied, which makes the derivative zero.